lua/mini/misc.lua

```diff
@@ -409,6 +409,137 @@ MiniMisc.zoom = function(buf_id, config)
   end
 end
 
+--- Restore the cursor on file reopen
+---
+--- What it does:
+--- - When reopening a file, this will make sure the cursor is placed back to
+---   the position where you left before.
+--- - This implements :help restore-cursor in a nicer way.
+--- - Creates autocommand for |BufReadPre| and |FileType| event with
+---   |MiniMisc.restore_cursor|
+---
+--- If you want to lazy load this use the |BufReadPre| event.
+---
+---@param opts table Allows to set options for |MiniMisc.restore_cursor|
+---   You can set the three options, `ignore_buftype` and `ignore_filetype` as
+---   an array of strings; `always_center` as an boolean.
+---   With `ignore_buftype` and `ignore_filetype` you can define buffer or file
+---   types which should be ignored and the cursor is then placed in the first
+--    line and column.
+--    The `center` option will always center the window, so the cursor
+--    is in the middle of the window. If you set it to `false` centering is
+--    turned off.
+---   Default:
+---   ```
+---   {
+---     ignore_buftype = { "quickfix", "nofile", "help" },
+---     ignore_filetype = { "gitcommit", "gitrebase" },
+---     center = true,
+---   }
+---   ```
```

Comment:
Could you, please, make it more like other `opts` documentations. Use [this](https://github.com/echasnovski/mini.nvim/blob/91017a96693408ef96efe9a2513c6ace0a87dc8d/lua/mini/ai.lua#L515-L524) as example.